### PR TITLE
Add System Locale layout direction for `Control` and `Window`

### PR DESCRIPTION
**Pull request by timothyqiu**
Currently, only the root node can use "Based on System Locale" as its layout direction (set via Project Settings). This difference creates complications when I try to simplify the logic of the layout directions.

This PR adds this option to other nodes (i.e., `Control` and `Window`).

In accommodate to this change, `LAYOUT_DIRECTION_LOCALE` is now deprecated in favor of `LAYOUT_DIRECTION_APPLICATION_LOCALE`.

### CODE REVIEW

File: scene/main/window.cpp [hunk 1]
```diff
@@ -2635,7 +2635,7 @@ void Window::set_unparent_when_invisible(bool p_unparent) {
 
 void Window::set_layout_direction(Window::LayoutDirection p_direction) {
 	ERR_MAIN_THREAD_GUARD;
-	ERR_FAIL_INDEX((int)p_direction, 4);
+	ERR_FAIL_INDEX((int)p_direction, 5);
```
Comment (bruvzg) [inline scene/main/window.cpp:2638]: We probably should have `LAYOUT_DIRECTION_MAX` to avoid setting it manually.

**Review by bruvzg — APPROVED** [review]

File: scene/main/window.cpp (hunk 1 above)
Comment (timothyqiu) [reply scene/main/window.cpp]: `LAYOUT_DIRECTION_MAX` added.

File: scene/gui/control.cpp [hunk 2]
```diff
@@ -3042,7 +3042,7 @@ void Control::set_layout_direction(Control::LayoutDirection p_direction) {
 	if (data.layout_dir == p_direction) {
 		return;
 	}
-	ERR_FAIL_INDEX((int)p_direction, 4);
+	ERR_FAIL_INDEX((int)p_direction, LAYOUT_DIRECTION_MAX);
```
Comment (Repiteo) [inline scene/gui/control.cpp:3045]:
```suggestion
	ERR_FAIL_INDEX(p_direction, LAYOUT_DIRECTION_MAX);
```

File: scene/main/window.cpp [hunk 3]
```diff
@@ -2635,7 +2635,7 @@ void Window::set_unparent_when_invisible(bool p_unparent) {
 
 void Window::set_layout_direction(Window::LayoutDirection p_direction) {
 	ERR_MAIN_THREAD_GUARD;
-	ERR_FAIL_INDEX((int)p_direction, 4);
+	ERR_FAIL_INDEX((int)p_direction, LAYOUT_DIRECTION_MAX);
```
Comment (Repiteo) [inline scene/main/window.cpp:2638]:
```suggestion
	ERR_FAIL_INDEX(p_direction, LAYOUT_DIRECTION_MAX);
```

File: scene/main/window.h [hunk 4]
```diff
@@ -86,9 +86,15 @@ class Window : public Viewport {
...
 		LAYOUT_DIRECTION_INHERITED,
-		LAYOUT_DIRECTION_LOCALE,
+		LAYOUT_DIRECTION_APPLICATION_LOCALE,
 		LAYOUT_DIRECTION_LTR,
-		LAYOUT_DIRECTION_RTL
+		LAYOUT_DIRECTION_RTL,
+		LAYOUT_DIRECTION_SYSTEM_LOCALE,
+		LAYOUT_DIRECTION_MAX,
+
+#ifndef DISABLE_DEPRECATED
```
Comment (Repiteo) [inline scene/main/window.h:95]:
```suggestion
		LAYOUT_DIRECTION_MAX,
#ifndef DISABLE_DEPRECATED
```

**Review by Repiteo — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**